Parsers/ASimNetworkSession/Parsers/ASimNetworkSessionCiscoMeraki.yaml

```diff
@@ -1,46 +1,580 @@
+        | extend temp_dstipport = coalesce(dst, dns_server)
+        | extend temp_dstipport =  iff(
+                                temp_dstipport has '"',
+                                replace_string(tostring(temp_dstipport), '"', ''),
+                                replace_string(tostring(temp_dstipport), "'", "")
+                            )
+        | parse temp_dstipport with * "[" temp_dstip "]:" temp_dstport
+        | extend DstIpAddr = iff(temp_dstipport has ".", split(temp_dstipport, ":")[0], coalesce(temp_dstip, temp_dstipport))
+        | extend DstPortNumber = toint(coalesce(dport, temp_dstport))
+        | extend DstPortNumber = toint(iff(isempty(DstPortNumber) and DstIpAddr has ".", split(temp_dstipport, ":")[1], DstPortNumber))
+        | extend SrcMacAddr = case(
+                            isnotempty(SrcMacAddr),
+                            SrcMacAddr,
+                            LogType == "airmarshal_events",
+                            src, 
+                            coalesce(client_mac, mac, shost)
+                        )
+        | extend SrcMacAddr =  iff(
+                            SrcMacAddr has '"',
+                            replace_string(tostring(SrcMacAddr), '"', ''),
+                            replace_string(tostring(SrcMacAddr), "'", "")
+                        )
+        | extend DstMacAddr = iff(LogType == "airmarshal_events", dst, dhost)
+        | extend DstMacAddr =  iff(
+                            DstMacAddr has '"',
+                            replace_string(tostring(DstMacAddr), '"', ''),
+                            replace_string(tostring(DstMacAddr), "'", "")
+                        )
+        | extend SrcIpAddr = iff(SrcIpAddr == SrcMacAddr, "", SrcIpAddr)
+        | extend DstIpAddr = iff(DstIpAddr == DstMacAddr, "", DstIpAddr)
+        | extend DvcMacAddr = column_ifexists("wired_mac", "")
+        | extend DvcMacAddr = iff(
+                            DvcMacAddr has '"',
+                            replace_string(tostring(DvcMacAddr), '"', ''),
+                            replace_string(tostring(DvcMacAddr), "'", "")
+                        )
+        | parse kind=regex Substring with * " VLAN " VLAN "[ ]{0,1}" restmessage
+        | parse kind=regex VLAN with VLAN2 " with " restmessage2
+        | parse kind=regex Substring with * "message: " message "[ ]{0,1}" restmessage
+        | extend
+            DvcAction = case(
+                    Substring has "allow",
+                    "Allow",
+                    Substring has "deny",
+                    "Deny",
+                    Substring has "Blocked",
+                    "Deny",
+                    ""
+                ),
+            EventMessage = iff(
+                    LogSubType has_any("Blocked DHCP server", "Virtual router collision"),
+                    Substring,
+                    coalesce(message, "")
+                ),
+            SrcUserName = column_ifexists("identity", ""),
```

Comment:
Since we are using parse-kv, "identity" field will always exists with or without the value. Column_ifexist is not required.

Reply:
Done